pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -0,0 +1,963 @@
+            },
+            "secureSettings": {
+              "integrationKey": "pagerduty_recv/pagerduty_test"
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+`
+
+var expAlertmanagerConfigFromAPI = `
+{
+  "template_files": null,
+  "alertmanager_config": {
+    "route": {
+      "receiver": "slack_recv1",
+      "group_wait": "0s",
+      "group_by": [
+        "alertname"
+      ],
+      "routes": [
+        {
+          "receiver": "email_recv",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"EmailAlert\""
+          ]
+        },
+        {
+          "receiver": "slack_recv1",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"SlackAlert1\""
+          ]
+        },
+        {
+          "receiver": "slack_recv2",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"SlackAlert2\""
+          ]
+        },
+        {
+          "receiver": "pagerduty_recv",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"PagerdutyAlert\""
+          ]
+        },
+        {
+          "receiver": "dingding_recv",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"DingDingAlert\""
+          ]
+        },
+        {
+          "receiver": "teams_recv",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"TeamsAlert\""
+          ]
+        },
+        {
+          "receiver": "webhook_recv",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"WebhookAlert\""
+          ]
+        },
+        {
+          "receiver": "telegram_recv",
+          "group_wait": "0s",
+          "group_by": [
+            "alertname"
+          ],
+          "matchers": [
+            "alertname=\"TelegramAlert\""
+          ]
+        }
+      ]
+    },
+    "templates": null,
+    "receivers": [
+      {
+        "name": "email_recv",
+        "grafana_managed_receiver_configs": [
+          {
+            "id": 0,
+            "uid": "",
+            "name": "email_test",
+            "type": "email",
+            "isDefault": false,
+            "sendReminder": false,
+            "disableResolveMessage": false,
+            "frequency": "",
+            "created": "0001-01-01T00:00:00Z",
+            "updated": "0001-01-01T00:00:00Z",
+            "settings": {
+              "addresses": "test@email.com",
+              "singleEmail": true
+            },
+            "secureFields": {}
```

Comment:
This is where I wanted to get on this PR, to me, this payload doesn't look right. I get the reason why it is this way though. In the configuration struct, the embedded field is a struct intended to be saved to the database - I think we should get away from this but I want to hear @papagian's thoughts on the matter and that we reach consensus here on what to do next.

However, I wouldn't expect the work to be done as part of this PR - An issue describing the problem/solution is more than acceptable.

Reply:
I am proposing removal of unwanted fields in this PR https://github.com/grafana/grafana/pull/34036